### PR TITLE
RCAL-704 Update resample step to handle the L3 meta data

### PR DESCRIPTION
**Pull request by stscieisenhamer**
Resolves [RCAL-704](https://jira.stsci.edu/browse/RCAL-704)

This PR addresses resolves updates the resample step for compatibility with the L3 metadata

Issues resolved:
- transfer of Level 2 meta info into Level 3

**Checklist**
- [x] added entry in `CHANGES.rst` under the corresponding subsection
- [x] updated relevant tests
- [x] updated relevant documentation
- [ ] updated relevant milestone(s)
- [x] added relevant label(s)
- [ ] ran regression tests, post a link to the Jenkins job below. [How to run regression tests on a PR](https://github.com/spacetelescope/romancal/wiki/Running-Regression-Tests-Against-PR-Branches)

### CODE REVIEW

**Comment by stscieisenhamer** [comment]
There are many unresolved issues that I will be listing in the [RCAL-704](https://jira.stsci.edu/browse/RCAL-704) issue.

**Comment by PaulHuwe** [comment]
The Level 3 Metadata PR has been updated to change the `pass` keyword to `pass_number`.

**Comment by stscieisenhamer** [comment]
This depends on roman_datamodels [PR#288](https://github.com/spacetelescope/roman_datamodels/pull/288) and rad [PR#334](https://github.com/spacetelescope/rad/pull/334/files), both of which I have PR's against.

Also dependent on rad [PR#351](https://github.com/spacetelescope/rad/pull/351) for WCS support.

File: romancal/resample/resample.py [hunk 1]
```diff
@@ -671,3 +669,85 @@ def drizzle_arrays(
...
+    basic.program: observation.program
+    basic.survey: obervation.survey
+    basic.optical_element: optical_element
+    basic.instrument: instrument.name
+    basic.telescope: telescope
+    program: program
+    """
+    l3_meta.basic.visit = l2_meta.observation.visit
+    l3_meta.basic.segment = l2_meta.observation.segment
+    # l3_meta.basic.pass = l2_meta.observation.pass
```
Comment (schlafly) [inline romancal/resample/resample.py:700]: I've forgotten; can we get this one in before merging?

File: romancal/resample/resample.py [hunk 2]
```diff
@@ -671,3 +669,85 @@ def drizzle_arrays(
...
+    # l3_wcsinfo.ra_corn1 = from bounding box
+    # l3_wcsinfo.ra_corn2 = from bounding box
+    # l3_wcsinfo.ra_corn3 = from bounding box
+    # l3_wcsinfo.ra_corn4 = from bounding box
+    # l3_wcsinfo.dec_corn1 = from bounding box
+    # l3_wcsinfo.dec_corn2 = from bounding box
+    # l3_wcsinfo.dec_corn3 = from bounding box
+    # l3_wcsinfo.dec_corn4 = from bounding box
+    # l3_wcsinfo.orientat = ???
+    # l3_wcsinfo.orientat_local = ???
```
Comment (schlafly) [inline romancal/resample/resample.py:753]: I guess we can fill these out in a later PR.  You might see if you can fill out some of them using the information in this file on central store, roman/eschlafly/tessellation/tessellation.py, though I am coming a bit from the opposite direction (trying to make a gwcs object) to some extent.

**Review by schlafly — COMMENTED** [review]
Thanks.  Remind me: this will break the resample regression test, and in the context of pass -> pass_number being a dependency will break a lot of other things.  But otherwise we think this should be fine?

**Comment by schlafly** [comment]
Can you get the unit tests working?

**Comment by stscieisenhamer** [comment]
Note: this code is dependent on PR in both rdm and rad, concerning L3 meta.

**Comment by PaulHuwe** [comment]
Passed regression tests, modulo an expected validation change (truth file requires updating):
https://plwishmaster.stsci.edu:8081/job/RT/job/Roman-Developers-Pull-Requests/580/

**Review by PaulHuwe — APPROVED** [review]
LGTM

**Review by ddavis-stsci — APPROVED** [review]
LGTM

File: pyproject.toml [hunk 3]
```diff
@@ -22,10 +22,10 @@ dependencies = [
...
     'pyparsing >=2.4.7',
     'requests >=2.26',
-    'rad >= 0.18.0',
-    # 'rad @ git+https://github.com/spacetelescope/rad.git@main',
-    'roman_datamodels >= 0.18.0',
-    # 'roman_datamodels @ git+https://github.com/spacetelescope/roman_datamodels.git@main',
+    # 'rad >= 0.18.0',
+    'rad @ git+https://github.com/spacetelescope/rad.git@main',
+    # 'roman_datamodels >= 0.18.0',
+    'roman_datamodels @ git+https://github.com/spacetelescope/roman_datamodels.git@main',
```
Comment (WilliamJamieson) [inline pyproject.toml:28]:
```suggestion
    'rad @ git+https://github.com/spacetelescope/rad.git',
    # 'roman_datamodels >= 0.18.0',
    'roman_datamodels @ git+https://github.com/spacetelescope/roman_datamodels.git',
```
Can you try making this change?

**Pull request merged**